transformation-scripts/semopenalex-keywords.py

```diff
@@ -0,0 +1,285 @@
+    keyword = clean_url(keyword)
+    keyword = "https://semopenalex.org/keywords/" + keyword
+    return keyword
+
+
+# info for namespaces used in SOA
+soa_namespace_class = "https://semopenalex.org/ontology/"
+
+# SOA classes used in this file
+soa_class_keyword = URIRef(soa_namespace_class + "Keyword")
+
+# SOA predicates
+
+# topics entity context
+context = URIRef("https://semopenalex.org/keywords/context")
+
+#topic scheme URI
+keyword_scheme_uri = URIRef("https://semopenalex.org/keywords/")
+
+i = 0
+error_count = 0
+keywords_graph = Graph(identifier=context)
+
+today = date.today()
+
+# Entity type of the snapshot files to be processed
+##########
+ENTITY_TYPE_INPUT = 'topics'
+##########
+
+##########
+ENTITY_TYPE_OUTPUT = 'keywords'
+##########
+
+data_dump_input_root_dir = '/opt/openalex-snapshot'
+
+absolute_path = os.path.dirname(__file__)
+trig_output_dir_path = os.path.join(absolute_path, '../graphdb-preload/graphdb-import/') 
+trig_output_file_path = f'{trig_output_dir_path}{ENTITY_TYPE_OUTPUT}-semopenalex-{today}.trig'
+
+data_dump_start_time = time.ctime()
+print('topics entity files started to download at: '+ data_dump_start_time)
+# Copy institutions entity snapshot
+client = boto3.client("s3", config=Config(signature_version=UNSIGNED))
+file_names, folders = get_file_folders(client, "openalex", "data/topics/")
+download_files(client, "openalex", data_dump_input_root_dir, file_names, folders)
+print('topics entity files finished to download.')
+
+start_time = time.ctime()
+print('topics entity started to transform at: '+ start_time)
+
+with open(trig_output_file_path, "w", encoding="utf-8") as g:
+
+    #Path where the OpenAlex data for the current entity type is located
+    data_dump_input_entity_dir = f'{data_dump_input_root_dir}/data/{ENTITY_TYPE_INPUT}/*'
+
+
+    # initialize and add domain scheme URI
+    keywords_graph.add((keyword_scheme_uri, RDF.type, SKOS.ConceptScheme))
+    keywords_graph.add((keyword_scheme_uri, SKOS.prefLabel, Literal("SemOpenAlex Keywords", datatype = XSD.string)))
+    keywords_graph.add((keyword_scheme_uri, URIRef("http://purl.org/dc/terms/description"), Literal("SemOpenAlex keywords are based on SemOpenAlex topics. They are specific words or phrases used to capture the essential topics or ideas of an entity.", datatype = XSD.string)))
+
+
+    for filename in glob.glob(os.path.join(data_dump_input_entity_dir, '*.gz')):
+        with gzip.open(filename, 'r') as f:
+            for line in f:
+                try:
+                    json_data = json.loads(line.decode('utf-8'))
+
+                    # keyword-URI
+                    # keywords
+                    topic_keywords = json_data['keywords']
+                    if not topic_keywords is None:
+                        for keyword in topic_keywords:
+                            replacements_url = clean_url(keyword)
+                            keyword_uri = transform_keyword_to_uri(keyword)
```

Comment:
maybe a quick check whether it is a valid IRI or not using `from rdflib import term` library.  For example, 
```
 if term._is_valid_uri(keyword_uri):

```

Reply:
Done.